ddl/ddl_api.go

```diff
@@ -636,6 +637,108 @@ func checkConstraintNames(constraints []*ast.Constraint) error {
 	return nil
 }
 
+func checkCreatePartitionNameUnique(part *model.PartitionInfo) error {
+	if part.Type != model.PartitionTypeRange {
+		return nil
+	}
+	newPars := part.Definitions
+	set := make(map[string]struct{})
+	for _, newPar := range newPars {
+		if _, ok := set[strings.ToLower(newPar.Name)]; ok {
+			return ErrSameNamePartition.GenByArgs(newPar.Name)
+		}
+		set[strings.ToLower(newPar.Name)] = struct{}{}
+	}
+	return nil
+}
+
+// checkCreatePartitionValue checks 'less than value' must be strictly increasing for each partition.
+func checkCreatePartitionValue(part *model.PartitionInfo) error {
+	if part.Type == model.PartitionTypeRange {
+		defs := part.Definitions
+		rangeValue := defs[0].LessThan[0]
+		if strings.EqualFold(rangeValue, "MAXVALUE") && len(defs) > 1 {
+			return errors.Trace(ErrPartitionMaxvalue)
+		}
+		currentRangeValue, err := strconv.Atoi(rangeValue)
+		if err != nil {
+			return errors.Trace(err)
+		}
+		if strings.EqualFold(defs[len(defs)-1].LessThan[0], "MAXVALUE") {
+			defs = defs[:len(defs)-1]
+		}
+
+		for i := 1; i < len(defs); i++ {
+			if strings.EqualFold(defs[i].LessThan[0], "MAXVALUE") && i != len(defs)-1 {
+				return errors.Trace(ErrPartitionMaxvalue)
+			}
+			nextRangeValue, err := strconv.Atoi(defs[i].LessThan[0])
+			if err != nil {
+				return errors.Trace(err)
+			}
+			if nextRangeValue <= currentRangeValue {
+				return errors.Trace(ErrRangeNotIncreasing)
+			}
+			currentRangeValue = nextRangeValue
+		}
+	}
+	return nil
+}
+
+func buildCreateTablePartitionInfo(ctx sessionctx.Context, d *ddl, s *ast.CreateTableStmt, cols []*table.Column) (*model.PartitionInfo, error) {
```

Comment:
Could we move these function to the file of "ddl/partition.go"? There is a lot of code in the current file.